src/components/chat/chat-panel.tsx

```diff
@@ -20,11 +20,14 @@ interface IProps {
 
 export const ChatPanel: React.FC<IProps> = ({ user, activeNavTab, focusDocument, focusTileId, onCloseChatPanel }) => {
   const document = useDocumentOrCurriculumMetadata(focusDocument);
+  const curriculumContent = useCurriculumContent(focusDocument);
+  const ordering = curriculumContent?.getTilesInDocumentOrder();
   const { data: comments } = useDocumentComments(focusDocument);
   const { data: unreadComments } = useUnreadDocumentComments(focusDocument);
   const documentComments = comments?.filter(comment => comment.tileId == null);
-  const tileComments = comments?.filter(comment => comment.tileId === focusTileId);
-  const postedComments = focusTileId ? tileComments : documentComments;
+  const allTileComments = comments?.filter(comment=> comment.tileId !== null);
+  const commentsInDocumentOrder = ordering && allTileComments ? allTileComments.sort((a: any, b: any) => ordering.indexOf(a.tileId) - ordering.indexOf(b.tileId)) : [];
```

Comment:
Fix line length lint warning:
```suggestion
  const compare = (a: any, b: any) => ordering.indexOf(a.tileId) - ordering.indexOf(b.tileId);
  const commentsInDocumentOrder = ordering && allTileComments ? allTileComments.sort(compare) : [];
```